trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,371 @@
+! This module defines and instantiates objects
+! for a hybrid persistence levelpool type
+! reservoir. The hybrid reservoir type
+! inherits input and output stucts from the
+! reservoir base module and calls instantiation
+! of these into sub-objects. The hybrid
+! reservoir type also points to types for
+! hybrid parameters and state and calls
+! instantiation of these into sub-objects.
+! A pointer to a levelpool reservoir object
+! is also held in state, and this module
+! instantiates that levelpool object. There
+! is also a subroutine to run hybrid reservoir
+! that is derived from the reservoir base
+! type interface to run reservoir.
+
+module module_persistence_levelpool_hybrid
+
+    use module_persistence_levelpool_hybrid_parameters, only: hybrid_parameters_interface
+    use module_persistence_levelpool_hybrid_state, only: hybrid_state_interface
+    use module_levelpool, only: levelpool_type
+    use module_reservoir_utilities, only: modify_for_projected_storage, warn_negative_inflow, &
+                                          create_hybrid_diagnostic_log_file, &
+                                          log_hybrid_diagnostic_data
+    use module_reservoir_base
+    use module_reservoir_read_timeslice_data, only: timeslice_data
+    implicit none
+
+    ! Extend/derive hybrid type from the abstract base
+    ! type for reservoirs.
+    type, extends(reservoir_base_type) :: persistence_levelpool_hybrid_type
```

Comment:
I would actually drop "base" and "type" and "struct". "base" implies type anyway.
It's fair if there is no inheritance from levelpool. But the naming convention does not reflect the inheritance, this is what I'm suggesting, along with dropping the "struct" or "type" self-declaration. "reservoir_levelpool_persistence_hybrid" achieves both. 

If there are some "abstract/archtypal" objects, like a "reservoir", then the fact that the variable name starts with one of those names is a better way of indicating an object compared to tagging the end of the variable "_type". If the abstract is "reservoir_" then all subclasses are "reservoir_something_or_other". At least in my opinion that's cleaner, more informative. I think it would be beneficial to develop such naming conventions for abstracts and subclasses to help improve the naming consistency and information content in the names.

Reply:
@jdmattern-noaa I will change these and drop those terms.